packages/lesswrong/lib/instanceSettings.ts

```diff
@@ -110,8 +111,8 @@ export class PublicInstanceSetting<SettingValueType> {
   Public Instance Settings
 */
 
-export type ForumTypeString = "LessWrong"|"AlignmentForum"|"EAForum";
-export const allForumTypes: Array<ForumTypeString> = ["LessWrong","AlignmentForum","EAForum"];
+export const allForumTypes = new TupleSet(["LessWrong","AlignmentForum","EAForum"]);
```

Comment:
```suggestion
export const allForumTypes = new TupleSet(["LessWrong","AlignmentForum","EAForum"] as const);
```

Otherwise `ForumTypeString` will be of type `string`

Reply:
Now that we're on Typescript 5, I believe we can change the definition of `TupleSet` from
```
export class TupleSet<T extends ReadonlyArray<string|number>> extends Set<string|number> {
```
to
```
export class TupleSet<const T extends ReadonlyArray<string|number>> extends Set<string|number> {
```
and then we won't need to write `as const` any more (I haven't tested it though).

Reply:
Hmm, passes typecheck, but doesn't build.

Reply:
Added to esbuild in https://github.com/evanw/esbuild/releases/tag/v0.17.5 , we're on 15.

Reply:
Still no workie, not clue why. Throwing in the towel. 🏳